### PR TITLE
Alerting: Round $values when using default String()

### LINKED ISSUE #67186: Alerting: Round $values when using default String()
## What happened:

When using the `$values` variable in custom annotations and labels, printing a value such as `{{ $values.B }}` calls the [`String()` method on `TemplateCaptureValue`](https://github.com/grafana/grafana/blob/v9.4.7/pkg/services/ngalert/state/template.go#L28). However, since it uses `strconv.FormatFloat(v.Value, 'f', -1, 64)` it causes values to be returned with much more precision than is required. For example:

```
The value is 0.007231050985307748
```

## What you expected to happen:

I would like to make the `String()` method return just the first two non-zero digits. For example:

```
The value is 0.0072
```

If we use fixed precision then alerts can contain unhelpful and confusing values, such as:

```
The value is 0.00
```

when the value is in fact non-zero.

## Environment:
- Grafana version:
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by sladyn98**
**What is this feature?**

Alerting: Round $values when using default String()

**Why do we need this feature?**

 Iwould like to make the String() method return just the first two non-zero digits. For example:

```
The value is 0.0072
```

If we use fixed precision then alerts can contain unhelpful and confusing values, such as:

```
The value is 0.00
```

The String() method in the templateCaptureValue struct has been optimized for performance and readability. Originally, the function worked by converting the float value to a string and then iterating over this string to find the position of the first non-zero digit. This was done to calculate the necessary precision for displaying two significant digits.

The updated version of the function uses mathematical operations to achieve the same result. It uses the math.Log10 function to calculate the position of the first non-zero digit, which is then used to calculate the necessary precision. This approach reduces the complexity of the code and should result in better performance, especially for large numbers.
**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #67186

**Special notes for your reviewer:**

Please check that:
- [ x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by sladyn98** [comment]
CC @grobinson-grafana @yuri-tceretian

File: pkg/services/ngalert/state/template/template_test.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func TestValueString(t *testing.T) {
 	}, {
 		name:     "1.0 is returned as integer value",
 		value:    Value{Value: 1.0},
-		expected: "1",
+		expected: "1.0",
```
Comment (sladyn98) [inline pkg/services/ngalert/state/template/template_test.go:51]: Since we would now require two decimal places the value expected here is 1.0

Comment (yuri-tceretian) [reply pkg/services/ngalert/state/template/template_test.go:51]: I wonder if we can improve the logic a bit and keep the number integer.

Comment (sladyn98) [reply pkg/services/ngalert/state/template/template_test.go]: Yeah lets modify the function to return an integer value

Comment (sladyn98) [reply pkg/services/ngalert/state/template/template_test.go]: Now it works without modifying the tests.

**Comment by sladyn98** [comment]
cc @grobinson-grafana

**Comment by grobinson-grafana** [comment]
Please also add some tests 🙂

**Comment by armandgrillet** [comment]
@sladyn98 Thank you for this PR, please update it following our review or we will have to close it.

**Pull request closed without merging**